File: flutter_sdk/lib/src/domain/models/audio/audio_data.dart
// Copyright (c) Yalochat, Inc. All rights reserved.

import 'package:equatable/equatable.dart';

class AudioData extends Equatable {
  final String fileName;
  final List<double> amplitudes;
  final List<double> amplitudesFilePreview;
  // The duration in milliseconds of the audio
  final int duration;

  AudioData({
    this.fileName = '',
    List<double>? amplitudes,
    List<double>? amplitudesFilePreview,
    this.duration = 0,
  }) : amplitudes = amplitudes ?? [],
       amplitudesFilePreview = amplitudesFilePreview ?? [];

  AudioData copyWith({
    String? fileName,
    List<double>? amplitudes,
    List<double>? amplitudesFilePreview,
    int? duration,
  }) {
    return AudioData(
      fileName: fileName ?? this.fileName,
      amplitudes: amplitudes ?? this.amplitudes,
      amplitudesFilePreview:
          amplitudesFilePreview ?? this.amplitudesFilePreview,
      duration: duration ?? this.duration,
    );
  }

  @override
  List<Object?> get props => [fileName, amplitudes, duration, amplitudesFilePreview];
}

File: android-sdk/sdk/src/commonMain/kotlin/com/yalo/chat/sdk/domain/model/AudioData.kt
// Copyright (c) Yalochat, Inc. All rights reserved.

package com.yalo.chat.sdk.domain.model

// amplitudesPreview is the compressed 48-point preview used for waveform display.
data class AudioData(
    val fileName: String = "",
    val amplitudes: List<Double> = emptyList(),
    val amplitudesPreview: List<Double> = emptyList(),
    val durationMs: Long = 0L,
)
